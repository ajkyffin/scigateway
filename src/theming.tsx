import { createMuiTheme } from '@material-ui/core/styles';
import { ThemeOptions, Theme } from '@material-ui/core/styles/createMuiTheme';

export interface UKRIThemeOptions extends ThemeOptions {
  ukri: {
    purple: string;
    blue: string;
    orange: string;
    green: string;
    grey: string;
  };
}

export interface UKRITheme extends Theme {
  ukri: {
    purple: string;
    blue: string;
    orange: string;
    green: string;
    grey: string;
  };
}

export const buildTheme = (): Theme => {
  const options: UKRIThemeOptions = {
    typography: {
      useNextVariants: true,
    },
    palette: {
      primary: {
        main: '#1D4F91', // dark blue
      },
      secondary: {
        main: '#63666A', //grey
      },
    },
    ukri: {
      purple: '#8C4799',
      blue: '#1D4F91',
      orange: '#C34613',
      green: '#008275',
      grey: '#63666A',
    },
  };

  return createMuiTheme(options);
};
