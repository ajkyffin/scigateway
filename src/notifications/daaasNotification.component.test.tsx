import React from 'react';
import { createShallow, createMount } from '@material-ui/core/test-utils';
import { NotificationWithStyles } from './daaasNotification.component';
import { Action } from 'redux';
import { buildTheme } from '../theming';
import { MuiThemeProvider } from '@material-ui/core';

const theme = buildTheme();

function createDaaasNotification(
  severity: string,
  message: string
): React.ReactElement {
  return (
    <MuiThemeProvider theme={theme}>
      <NotificationWithStyles
        message={message}
        severity={severity}
        index={0}
        dismissNotification={(): Action => ({ type: 'test' })}
      />
    </MuiThemeProvider>
  );
}

describe('Daaas Notification component', () => {
  let shallow;
  let mount;

  beforeEach(() => {
    shallow = createShallow({});
    mount = createMount();
  });

  afterEach(() => {
    mount.cleanUp();
  });

  it('Daaas Notification success message renders correctly', () => {
    const wrapper = shallow(
      createDaaasNotification('success', 'success message')
    );
    expect(wrapper.dive().dive()).toMatchSnapshot();
  });

  it('Daaas Notification warning message renders correctly', () => {
    const wrapper = shallow(
      createDaaasNotification('warning', 'warning message')
    );
    expect(wrapper.dive().dive()).toMatchSnapshot();
  });

  it('Daaas Notification error message renders correctly', () => {
    const wrapper = shallow(createDaaasNotification('error', 'error message'));
    expect(wrapper.dive().dive()).toMatchSnapshot();
  });

  it('an action is fired when Daaas Notification button is clicked', () => {
    const mockDismissFn = jest.fn();

    const wrapper = mount(
      <MuiThemeProvider theme={theme}>
        <NotificationWithStyles
          message={'warning message'}
          severity={'warning'}
          index={0}
          dismissNotification={mockDismissFn}
        />
      </MuiThemeProvider>
    );

    wrapper.find('button').simulate('click');

    expect(mockDismissFn.mock.calls.length).toEqual(1);
  });
});
