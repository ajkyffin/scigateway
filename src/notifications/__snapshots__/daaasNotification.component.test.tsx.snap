// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Daaas Notification component Daaas Notification error message renders correctly 1`] = `
<div
  className="DaaasNotification-root-1"
>
  <pure(HighlightOffIcon)
    className="DaaasNotification-severityIcon-4"
  />
  <WithStyles(Typography)
    variant="body2"
  >
    error message
  </WithStyles(Typography)>
  <WithStyles(IconButton)
    className="DaaasNotification-button-2"
    onClick={[Function]}
  >
    <pure(ClearIcon)
      className="DaaasNotification-deleteIcon-3"
    />
  </WithStyles(IconButton)>
</div>
`;

exports[`Daaas Notification component Daaas Notification success message renders correctly 1`] = `
<div
  className="DaaasNotification-root-1"
>
  <pure(CheckCircleIcon)
    className="DaaasNotification-severityIcon-4"
  />
  <WithStyles(Typography)
    variant="body2"
  >
    success message
  </WithStyles(Typography)>
  <WithStyles(IconButton)
    className="DaaasNotification-button-2"
    onClick={[Function]}
  >
    <pure(ClearIcon)
      className="DaaasNotification-deleteIcon-3"
    />
  </WithStyles(IconButton)>
</div>
`;

exports[`Daaas Notification component Daaas Notification warning message renders correctly 1`] = `
<div
  className="DaaasNotification-root-1"
>
  <pure(ErrorOutlineIcon)
    className="DaaasNotification-severityIcon-4"
  />
  <WithStyles(Typography)
    variant="body2"
  >
    warning message
  </WithStyles(Typography)>
  <WithStyles(IconButton)
    className="DaaasNotification-button-2"
    onClick={[Function]}
  >
    <pure(ClearIcon)
      className="DaaasNotification-deleteIcon-3"
    />
  </WithStyles(IconButton)>
</div>
`;
